### DISCUSSION TITLE
Schematic for the board

### DISCUSSION THREAD

**Discussion by paulsteigel** [ArminJo/JK-BMSToPylontechCAN#11] [General]
Hi ArminJo,
Thank for your great project, I am trying to redraw the schematic from your diagram. Can you be so kind, checking this for me with thanks. You can access the project at https://easyeda.com/editor#project_id=1c0fef1a71f94c4e840e316561cc5b01
Yours very faithfully
Dang Dinh Ngoc
![image](https://github.com/ArminJo/JK-BMSToPylontechCAN/assets/8839597/7e88637a-23e3-4d5d-89f0-cc28528db689)

**Comment by ArminJo** [comment]
Great!
D2 is connected to the "Page" push button not 2516 INT.
CS is connected to D9 (by default).
D32 is connected to the "Debug" push button .
Vin of nano is connected to 6.6 V from battery.
BMS RX is connecter to D4  (by default).
BMS TX is connected to RXD.

I think, thats it.
Can I include the graphic and a link to it in my Readme?

**Reply by paulsteigel** [reply]
Yes, let's me redraw and send you the updated schematic!
One tip, consuming power directly from cell will create stress on JK-BMS - I would recommend using a DC-DC Buck converter 5-80V from the VBAT and ground will be of help and avoiding problem to the BMS!
Since I am on the rush for buying material to test today, can you login EasyEDA with your gmail, I can share with your gmail directly (if you already have one account). You can quickly update it without any difficulties!
Best -

**Reply by ArminJo** [reply]
Sharing is a good idea.
My EasyEDA account is ArminJo.
Which buck converter do you recommend, most are only up to 48 V.

**Comment by paulsteigel** [comment]
I have done the perfboard welding but the firmware uploading to arduino nano did not work! Have any idea on how to solve it?
avrdude: stk500_recv(): programmer is not responding
avrdude: stk500_getsync() attempt 10 of 10: not in sync: resp=0x97
Problem uploading to board.  See https://support.arduino.cc/hc/en-us/sections/360003198300 for suggestions.

**Reply by ArminJo** [reply]
Disconnect the BMS from the Nano RX line (or use a diode in the right direction).

**Reply by paulsteigel** [reply]
I have to do the following:
1. Downgrade the board version to 1.6.20
2. Change the processor
And it work! for now I am testing the connection with JK board!
I forgot the Diode for removing parasitic power from CAN, hop this still work!

**Reply by paulsteigel** [reply]
Hi Armin,
I made the board but I could not get the serial data from JK-BMS (mine is: JK-BD6A24s6p) to arrive! I tried 2 options:
1. Connect the the board (made in pegboard) to the BMS via the middle port (see photo 1)
![image](https://github.com/ArminJo/JK-BMSToPylontechCAN/assets/8839597/4d4d578a-03d7-45f6-ab82-46cde34ce8eb)
No serial data were logged, I turn to the 2nd
2. Connect a single nano to the BMS and tried logging data but nothing logged also
![image](https://github.com/ArminJo/JK-BMSToPylontechCAN/assets/8839597/b14a491c-f34b-495f-b8f5-1ed0a625d69a)
With a bit of learning regarding the JK-BMS's RS485 (customise), I found that it is a 3.3V logic level. Does this means I have to use a UART TTL converter?
Please advice me with thanks!

**Reply by ArminJo** [reply]
Ground connection?

**Reply by paulsteigel** [reply]
Hi Armin,
I used Uno alone for testing data receive, with just Tx pin from BMS connected to arduino Uno Pin 10 as RX using software serial and see some data arrive
![image](https://github.com/ArminJo/JK-BMSToPylontechCAN/assets/8839597/35781b88-4f28-49c4-a6db-33ba860c864b)
Regarding your previous response, where does the ground will be connected to the arduino? GND pin on arduino? I am powering Uno/nano from my labtop! I am a hobbyist so I thought I have GND (B-) connected to my board (the pegboard) so this would be common ground for all! Please be patient on my limited knowledge with thanks!
Please advice!
Ngoc

**Reply by paulsteigel** [reply]
Hi Armin, I finally have Nano worked with GND connected as you said and comment the testing line from your code and data get in as expected!
All the best. The schematic will have to revise a bit as your diode is placed differently from the first image comparing to the second one (with 1.25mm connector), Also the connection to Rx (on nano) will have to be revised too!

**Reply by ArminJo** [reply]
> your diode is placed differently from the first image comparing to the second one

already fixed 😀

**Reply by paulsteigel** [reply]
Please use this image for schematic with thanks.
![image](https://github.com/ArminJo/JK-BMSToPylontechCAN/assets/8839597/741d3036-e734-4fbe-8635-64cbb7f3fe25)
Tomorrow I will start wire up!
Great thanks to your valuable endeavor to help the community!
Ngoc - From Vietnam!

**Reply by ArminJo** [reply]
Thanks, but please replace the 1N4007 by a BAT85 schottky diode (different symbol)
and J1 and J2 of U5 (better MCP2515 CAN bus module) are connected / bridged for end termination.
I put your schematic in the readme now 😀

**Reply by paulsteigel** [reply]
Many thanks ArminJo, I will base on your schematic and draw a pegboard version first, then PCB for easy work! One question:
I have GND connected to B- why do i need GND from the JST 1.25mm connector? I measured and see that they are connected in common?

**Reply by paulsteigel** [reply]
Armin, I replaced the Diode with your recommended one (it's a Zener?). Also I dont put buzzer and buttons... and I2C connection to the LCD. Cos this is a minimum version for sending data to Inverter. Today I will try the connection to Luxpower to see how it work (Luxpower can connect with Pylontech battery). I am not quite sure zener will work
![image](https://github.com/ArminJo/JK-BMSToPylontechCAN/assets/8839597/d764ec66-e3d6-4ad2-930a-e0115f765b5c)
I dont really get your point of "and J1 and J2 of U5 (better MCP2515 CAN bus module)"- Please advice!
Put all onto a pegboard seems not working quite nice! I am trying a PCB version for now to see how it work! Currently, I am using the version with dupont connections (see image) - a bit ugly but worked (at least as it shown on serial monitor on my laptop).
I am trying to remove LCD, button...etc from the code to make it smaller, or if you have time, can you make it?
I checked the datasheet and found out that nano 5V can supply max 150mA and this is enough for the MPC2515. So no need a 7805 regulator! Let's see how it goes!
All in a small box
![image](https://github.com/ArminJo/JK-BMSToPylontechCAN/assets/8839597/4d58b19e-0ab5-4708-810c-938b8d24fa5d)
![image](https://github.com/ArminJo/JK-BMSToPylontechCAN/assets/8839597/1b7d00eb-d8f3-42c2-8e11-79c1a86e4bbc)

![image](https://github.com/ArminJo/JK-BMSToPylontechCAN/assets/8839597/7fb6597f-ed4f-4bbb-95a9-de6568e8996d)

**Reply by ArminJo** [reply]
Can you name "U5" with "MCP2515 CAN bus module"
Connect the jumpers J1 and J2 in your schematic with a wire.
My Nano 5V regulator was getting hot with LCD and CAN attached.
Can you remove the INT connection (pentagon) from the CAN module, it is only for CAN receiving.
What is the reason for making code smaller?

**Reply by paulsteigel** [reply]
Hi, I have updated the schematic accordingly!
Regarding code, some people might want to not having LCD, button etc (leave it there would be ok) and having code at minimum size (remove some not needed lib can be easy for following). Anyhow, it does not a matter, I am using your code with commented testing and debugging for quick viewing of the data presentation on Serial monitoring.
![image](https://github.com/ArminJo/JK-BMSToPylontechCAN/assets/8839597/106c83dc-3a6e-4f67-a139-7f608af1c77d)
For anyone need pegboard welding (size of: WxH of 70x50mm), please see this:
**Bottom**
![image](https://github.com/ArminJo/JK-BMSToPylontechCAN/assets/8839597/55d0aceb-74d6-406f-8214-ee51eb6ee3e5)
**Top**
![image](https://github.com/ArminJo/JK-BMSToPylontechCAN/assets/8839597/d998d68e-a805-450d-b414-73ffab433259)

**Comment by paulsteigel** [comment]
Armin, one question please. 
I connect can bus (2 wire H/L) only, I use a CAT5 RJ45 connecter (one end cut and keep only wire 3-L and 4-H), and connect to the inverter (luxpower SNA5000, they announce the Inverter supports Pylontech), however the inverter can not recognize the battery. 
Did I do anything wrong?

**Reply by ArminJo** [reply]
Do not forget ground!

**Reply by paulsteigel** [reply]
Hi Armin,
Thanks for your response, apparently, LuxPower has pylontech on the lis
https://luxpowertek.com/wp-content/uploads/2022/04/Luxpower-compatible-list_20220414.pdf
However, according to the CAN wiring layout,  it requires only: CANH and CANL
![image](https://github.com/ArminJo/JK-BMSToPylontechCAN/assets/8839597/90805510-8a12-45bc-b118-4e474153f577)
I checked multiple place on how to set up CAN wiring using RJ45, but CAN layouts established differently.
Further search on forums, some guys inform a different view on this with following layout (I am not testing yet)
![image](https://github.com/ArminJo/JK-BMSToPylontechCAN/assets/8839597/84b69c56-322c-4376-b124-dfef72db6f88)

Can you advice me on how to set up ground with thanks!
Ngoc!

**Reply by paulsteigel** [reply]
Hi ArminJo,
For some reason the nano version worked for me on only the ready version (welded by manufacture). With the Version that I have to weld jumper myself the "buffer timeout" happened despite of successful connection of MCP2515 (I test serial connection between my welded nano and other arduino, they are successfully connected and sending out/ receiving of data via RX/TX). However with Uno R3, querying and receiving UART data from JK-BMS was successfully done!. 
Finally, I decided to move on with Uno! I also created a PCB version for testing and hope it will work as well!
Thanks for all your guidance and support.
Great to have opportunity to apply your and other hobbyists' great works
Ngoc!
